### DISCUSSION TITLE
Are two implementations of documentation comment ID generator necessary?

### DISCUSSION THREAD

**Discussion by svick** [dotnet/roslyn#48323] [Q&A]
There are two ways to get a documentation comment ID for an `ISymbol`:

1. `DocumentationCommentId.CreateDeclarationId`/`DocumentationCommentId.CreateReferenceId`
2. `ISymbol.GetDocumentationCommentId`

These two approaches have separate implementations, for example:

https://github.com/dotnet/roslyn/blob/374ee3184a64ca07867320668ec4b1162daa2aa4/src/Compilers/Core/Portable/DocumentationCommentId.cs#L418-L441

https://github.com/dotnet/roslyn/blob/374ee3184a64ca07867320668ec4b1162daa2aa4/src/Compilers/CSharp/Portable/DocumentationComments/DocumentationCommentIDVisitor.PartVisitor.cs#L99-L123

Is it necessary to have two separate implementations that do very similar things, or could they be consolidated into one?

Note that even on this small sample, some differences are visible:

* One uses `MS.CA.IMethodSymbol`, the other `MS.CA.CS.Symbols.MethodSymbol`.
* The condition for when return type is output are different. E.g. for `int M()` in class `C`, one returns `M:C.M~System.Int32`, the other one just `M:C.M`. Even if such differences are intentional and should be preserved, consolidating the two implementations could still make sense.

**Comment by CyrusNajmabadi** [comment]
Definitely seems odd.  Would prefer we unify on the form used by the compiler today.
